client/src/accounts/authentication/register.hook.test.tsx

```diff
@@ -0,0 +1,107 @@
+import { describe, expect, it } from 'vitest';
+import { rest, server } from '~/mocks/server';
+import { act, renderHook, waitFor } from '~/test/utils';
+import { useRegister } from './register.hook';
+
+const ENDPOINT = '*/api/accounts/register/';
+
+interface Result {
+  id: Number;
+  error?: { message: string };
+  mutate: (form: Record<string, unknown>) => void;
+  isError: boolean;
+  isSuccess: boolean;
+  data?: Record<string, unknown>;
+}
+
+describe('useRegisterHook', () => {
+  it('should throw an error if any fields are missing', async () => {
+    // we're missing the confirm password
+    const expectedMessage = 'Confirm Password field is empty';
+    server.use(
+      rest.post(ENDPOINT, (req, res, ctx) => res(ctx.status(400), ctx.json({ error: { message: expectedMessage } }))),
+    );
+    const registerForm = {
+      email: 'bob@example.com',
+      firstName: 'Bob',
+      lastName: 'Ross',
+      password: 'otherpassword',
+    };
+    const { result } = renderHook<Result, unknown>(() => useRegister());
```

Comment:
Missing type for the form itself, this can be retrieved from the hook and your `registerForm` should adhere to it. I know it is a headache, but we gotta start writing TypeScript, not JavaScript.